senatus/src/index.ts

```diff
@@ -14,5 +17,33 @@ export default {
    * This gives you an opportunity to set up your data model,
    * run jobs, or perform some special logic.
    */
-  bootstrap(/*{ strapi }*/) {},
+  async bootstrap({ strapi }: { strapi: Strapi }) {
+    console.log("Senatus is bootstrapping");
+    console.log(strapi.config.get("server.app.rabbitUrl"));
+    const { 
+      subscribe,
+      publish,
+    } = await connect(strapi.config.get("server.app.rabbitUrl"));
+    console.log("Connected to RabbitMQ!");
+    await subscribe("results", async (msg) => {
+      const parsed = JSON.parse(msg);
+      const submission = await strapi.entityService.findOne('api::submission.submission', parsed.metadata.submissionId, {
+        populate: ['user', 'problem', 'language'],
+      });
+
+      for (const result of parsed.result) {
+        console.log("Updating execution", result.metadata.executionId);
+        const execution = await strapi.entityService.update('api::execution.execution', result.metadata.executionId, {
+          data: {
+            stdout: result.stdout,
+            stderr: result.stderr,
+            executionTime: result.time,
+            processed: true,
+            processedAt: new Date(),
+            publishedAt: new Date(),
```

Comment:
Why are these two identical?

Reply:
PublishedAt is the Strapi way of publishing an entity, processed at is our own indicator for when the execution actually finished.